internal/osbuild/assembler_test.go

```diff
@@ -0,0 +1,230 @@
+			args: args{
+				data: []byte(`{"options":{"bar":null}}`),
+			},
+			wantErr: true,
+		},
+		{
+			name: "qemu assembler empty",
+			assembler: Assembler{
+				Name:    "org.osbuild.qemu",
+				Options: &QEMUAssemblerOptions{},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.qemu","options":{"format":"","filename":"","size":0,"ptuuid":"","pttype":"","partitions":null}}`),
+			},
+		},
+		{
+			name: "qemu assembler full",
+			assembler: Assembler{
+				Name: "org.osbuild.qemu",
+				Options: &QEMUAssemblerOptions{
+					Format:   "qcow2",
+					Filename: "disk.qcow2",
+					Size:     2147483648,
+					PTUUID:   "0x14fc63d2",
+					PTType:   "mbr",
+					Partitions: []QEMUPartition{QEMUPartition{
+						Start:    2048,
+						Bootable: true,
+						Filesystem: QEMUFilesystem{
+							Type:       "ext4",
+							UUID:       "76a22bf4-f153-4541-b6c7-0332c0dfaeac",
+							Label:      "root",
+							Mountpoint: "/",
+						},
+					}},
+				},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.qemu","options":{"format":"qcow2","filename":"disk.qcow2","size":2147483648,"ptuuid":"0x14fc63d2","pttype":"mbr","partitions":[{"start":2048,"bootable":true,"filesystem":{"type":"ext4","uuid":"76a22bf4-f153-4541-b6c7-0332c0dfaeac","label":"root","mountpoint":"/"}}]}}`),
+			},
+		},
+		{
+			name: "tar assembler empty",
+			assembler: Assembler{
+				Name:    "org.osbuild.tar",
+				Options: &TarAssemblerOptions{},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.tar","options":{"filename":"","size":0}}`),
+			},
+		},
+		{
+			name: "tar assembler full",
+			assembler: Assembler{
+				Name: "org.osbuild.tar",
+				Options: &TarAssemblerOptions{
+					Filename:    "root.tar.xz",
+					Size:        0,
+					Compression: "xz",
+				},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.tar","options":{"filename":"root.tar.xz","size":0,"compression":"xz"}}`),
+			},
+		},
+		{
+			name: "rawfs assembler empty",
+			assembler: Assembler{
+				Name:    "org.osbuild.rawfs",
+				Options: &RawFSAssemblerOptions{},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.rawfs","options":{"filename":"","root_fs_uuid":"00000000-0000-0000-0000-000000000000","size":0}}`),
+			},
+		},
+		{
+			name: "rawfs assembler full",
+			assembler: Assembler{
+				Name: "org.osbuild.rawfs",
+				Options: &RawFSAssemblerOptions{
+					Filename:           "filesystem.img",
+					RootFilesystemUUID: uuid.MustParse("76a22bf4-f153-4541-b6c7-0332c0dfaeac"),
+					Size:               2147483648,
+				},
+			},
+			args: args{
+				data: []byte(`{"name":"org.osbuild.rawfs","options":{"filename":"filesystem.img","root_fs_uuid":"76a22bf4-f153-4541-b6c7-0332c0dfaeac","size":2147483648}}`),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			assembler := &tt.assembler
+			var gotAssembler Assembler
+			err := gotAssembler.UnmarshalJSON(tt.args.data)
```

Comment:
ping about args.data, see comment from yesterday.

Reply:
I don't think there is a better way around this. I have to loop through all of the test scenarios and compare the expected and actual results. I could use something like `tests[i].assembler`, but IMO it wouldn't look nicer that what I'm using now (as I found out, the code is partially generated by VS code, and I borrowed the base for the assembler tests from an existing test file).

Reply:
The original question was:
> Why do we need a proxy data type which encapsulates the .data field instead of providing .data (or whatever the name) directly inside tests ?

So I was asking why do we need to have `tt.args.data` instead of simply `tt.data`, iow why do we need the intermediate `args` struct which contains only 1 attribute inside of it? Why not assign the byte array directly as an attribute in the `tests` struct ?

Reply:
Now I see what you were getting at :), of course all you said makes perfect sense. Fixed as well.